### DISCUSSION TITLE
Recurrent neural network

### DISCUSSION THREAD

**Discussion by wangjiawen2013** [tracel-ai/burn#2445] [Q&A] [answered]
Hi,
Does burn provide recurrent neural network now ?

**Comment by laggui** [comment]
We've have [RNNs](https://burn.dev/burn-book/building-blocks/module.html#rnns) for a while now. `Gru` and `Lstm`/`BiLstm` more precisely.

Are you looking for something else specifically?

**Comment by wangjiawen2013** [comment]
In fact, I'm trying to adopt burn in my daily work. There are lots of artifical neural networks now (https://www.geeksforgeeks.org/types-of-neural-networks/) and I am interested in Generative Adversarial Networks (GAN), so I wonder if burn is well-established.
![image](https://github.com/user-attachments/assets/36f8e998-fe23-4139-93f7-78401ca971bc)
The types in the red rectangle have already been included in burn. What about others, do you have a roadmap ?

**Reply by nathanielsimard** [reply]
Burn is a dynamic neural network framework. You can build pretty much any neural network modules with it, but we may not have premade implementations for all kinds of neural networks.

**Comment by laggui** [comment, marked as answer]
We have transformer blocks. Autoencoders, as with a lot of the other networks in the list, are just combinations of layers that we already support (e.g., convolutions) with a specific structure and loss.

Spiking neural networks I am not very familiar with but I think that's an entirely different type of application which might not be suited to what we offer.

**Comment by wangjiawen2013** [comment]
Hi, Could you tell where to find autoencoders and generative adversarial network examples written in burn ?

**Reply by laggui** [reply]
We don't have any official examples, but I remember a community member posted this VAE implementation on discord a while back: https://github.com/code-cp/vae-burn
